client/pages/home.jsx

```diff
@@ -1,17 +1,174 @@
-
 import React from 'react';
 import Redirect from '../components/redirect';
 import AppContext from '../lib/app-context';
+import CustomAccordion from '../components/accordion';
+import EditForm from '../components/edit-form';
 
 export default class Home extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      caption: '',
+      style: '',
+      skill: '',
+      instrument: '',
+      mainInterest: '',
+      interest: '',
+      band: '',
+      about: '',
+      profileUrl: '',
+      isEditing: false
+
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.fileInputRef = React.createRef();
+    this.clickToEdit = this.clickToEdit.bind(this);
+  }
+
+  componentDidMount() {
+    if (this.context.user) {
+
+      const userId = this.context.user.userId;
+      fetch(`/api/profile/users/${userId}`, {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      })
+        .then(response => response.json())
+        .then(data => {
+
+          const { avaterUrl, avaterCaption, userStyle, userSkills, userInstruments, userPrimaryInterest, userInterest, userBand, userBio } = data;
+          this.setState({
+            caption: avaterCaption,
+            style: userStyle,
+            skill: userSkills,
+            instrument: userInstruments,
+            mainInterest: userPrimaryInterest,
+            interest: userInterest,
+            band: userBand,
+            about: userBio,
+            profileUrl: avaterUrl
+          });
+
+        })
+        .catch(error => {
+          console.error('Error:', error);
+        });
+    }
+  }
+
+  clickToEdit() {
+    this.setState({ isEditing: !this.state.isEditing });
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  }
+
+  handleSubmit(event) {
+    const userId = this.context.user.userId;
+    event.preventDefault();
+
+    const { caption, style, skill, instrument, mainInterest, interest, band, about } = this.state;
+    const userInfo = {
+      caption: caption,
+      style: style,
+      skill: skill,
+      instrument: instrument,
+      mainInterest: mainInterest,
+      interest: interest,
+      band: band,
+      about: about
+    };
+
+    const form = new FormData();
+
+    fetch(`/api/profile/users/${userId}`, {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json'
+      },
```

Comment:
Send this with the token instead of `${userId}` in the URL. Check `api-authorization` exercise.